docs/arcaflow/workflows/flow-control.md

```diff
@@ -1,14 +1,75 @@
 # Using Flow Control Mechanics
 
-Flow control allows the workflow author to build a workflow with a decision tree based on supported flow logic. These flow control operations are not implemented by plugins, but are part of the workflow engine itself.
+Flow control allows the workflow author to build a workflow with a decision tree based
+on supported flow logic. These flow control operations are not implemented by plugins,
+but are part of the workflow engine itself.
+
+## Implicit Step Relationships
+
+Any time the input of a step relies on the output of another step via an [Arcaflow
+expression](expressions.md), an implicit step relationship is established. In this case,
+the Arcaflow engine holds the execution of the dependent step until the output from the
+supplier step is available.
+
+```yaml title="workflow.yaml"
+version: v0.2.0
+steps:
+  step_a:
+    plugin: 
+      deployment_type: image
+      src: quay.io/some/container/image
+    input:
+      some:
+        key: !expr $.input.some_value
+  step_b:
+    plugin: 
+      deployment_type: image
+      src: quay.io/some/container/image
+    input:
+      some:
+        key: !expr $.steps.step_a.outputs.success.some_value
+```
+
+## Explicit Step Relationships
+
+Sometimes it is important to serialize workflow steps even if they do not have a data
```

Comment:
To make it clear what we mean by serialize the steps, I would say "serialize the order of workflow steps" instead.